data-prepper-plugins/opensearch/README.md

```diff
@@ -165,6 +165,8 @@ If a single record turns out to be larger than the set bulk size, it will be sen
 
 - `document_root_key`: The key in the event that will be used as the root in the document. The default is the root of the event. If the key does not exist the entire event is written as the document. If the value at the `document_root_key` is a basic type (ie String, int, etc), the document will have a structure of `{"data": <value of the document_root_key>}`. For example, If we have the following sample event: 
 
+- `tags_key_name` (optional): The key name to be used to include event tags in the document. By default this is not set. When this option is not used, tags are not included in the document. If the option is used but the event does not have any tags, it stored as empty list in the document.
```

Comment:
We did something similar with routes which made the routes available as a global property.

Here is some of the work for it.

https://github.com/opensearch-project/data-prepper/blob/59c48e29a25e15b16e38f4c118477a041e458121/data-prepper-api/src/main/java/org/opensearch/dataprepper/model/configuration/SinkModel.java#L69-L71

I think that we could have Data Prepper core read this flag and then update the underlying Event for the sink. Then sinks don't have to have any tagging-specific code. Thoughts?

Reply:
Thanks @dlvenable and @graytaylor0. It makes sense to make this a sink level `JsonProperty`. I will make it similar to `routes` sink option.

Reply:
@dlvenable reading the flag can be done in the Data Prepper core but the flag needs to be set in each Sink separately because not all sinks will have this option set. Secondly, we cannot update underlying event (even the current approach I have here is incorrect) because modifying an event would automatically modify the event FOR all sinks because we do not make a copy today when there are non-pipeline-connector sinks. 

Only way to do it without copying is to do `event.toJsonStringWithTags()` instead of `event.toJsonString()`  in `DocumentBuilder.build` in OpenSearchSink (and equivalent code in other sinks)